subsys/net/ip/net_tls.c

```diff
@@ -0,0 +1,733 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_TLS_DEBUG)
+#define SYS_LOG_DOMAIN "net/tls"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <stdio.h>
+#include <sys/param.h>
+
+#include "net_private.h"
+#include "net_tls_internal.h"
+#include "tcp_internal.h"
+
+#define TIMEOUT_TLS_RX_MS       100
+#define TIMEOUT_TLS_TX_MS       100
+
+#if defined(CONFIG_NET_PRECONFIGURE_TLS_CREDENTIALS)
+#include CONFIG_NET_TLS_CREDENTIALS_FILE
+#endif
+
+static mbedtls_ctr_drbg_context tls_ctr_drbg;
+
+struct net_tls_credential {
+	enum net_tls_credential_type type;
+	sec_tag_t tag;
+	const void *buf;
+	size_t len;
+};
+
+static struct net_tls_credential credentials[CONFIG_NET_MAX_CREDENTIALS_NUMBER];
```

Comment:
I'm not sure I get this well. If this a global static pool of "credentials" (what is credential in this context?), shared among all TLS contexts, then well, there should be a big fat comment that it is that ;-).

Reply:
That's exactly what it is.